### PR TITLE
Added BST search insert and delete

### LINKED ISSUE #404: [Feature Request] Insert Delete Search in a BST
## 🚀 Feature Request
I want to add three problems in the Binary Search Tree Section 

### Describe the solution you'd like
I will add insert, search and delete in a BST.

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by haseebzaki-07**
### Description
Added BST search insert and delete operations in the Binary Search Tree Section
Fixes #404 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/binary-search-tree/delete-BST.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    root->left->left = new TreeNode(1);
+    root->left->right = new TreeNode(3);
+
+    int valueToDelete = 2;
+    root = deleteBST(root, valueToDelete);
+
+    cout << "Value " << valueToDelete << " deleted from the BST." << endl;
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/binary-search-tree/delete-BST.md:106]: close your code example with ```` ``` ````

File: docs/binary-search-tree/insert-BST.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    root->left->left = new TreeNode(1);
+    root->left->right = new TreeNode(3);
+
+    int valueToInsert = 5;
+    root = insertBST(root, valueToInsert);
+
+    cout << "Value " << valueToInsert << " inserted in the BST." << endl;
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/binary-search-tree/insert-BST.md:75]: close your code example with ```` ``` ````

File: docs/binary-search-tree/search-BST.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    TreeNode* result = searchBST(root, target);
+
+    if (result != NULL) {
+        cout << "Found: " << result->val << endl;
+    } else {
+        cout << "Value not found in the BST." << endl;
+    }
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/binary-search-tree/search-BST.md:74]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestion

**Comment by haseebzaki-07** [comment]
hey @ajay-dhangar I fixed the ending of the codes.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**